evaluator.plandefinition/src/main/java/org/opencds/cqf/cql/evaluator/plandefinition/r4/PlanDefinitionProcessor.java

```diff
@@ -0,0 +1,725 @@
+
+  public PlanDefinitionProcessor(FhirContext fhirContext, FhirDal fhirDal, LibraryProcessor libraryProcessor, ExpressionEvaluator expressionEvaluator,
+      ActivityDefinitionProcessor activityDefinitionProcessor, OperationParametersParser operationParametersParser) {
+    requireNonNull(fhirContext, "fhirContext can not be null");
+    requireNonNull(fhirDal, "fhirDal can not be null");
+    requireNonNull(libraryProcessor, "LibraryProcessor can not be null");
+    requireNonNull(operationParametersParser, "OperationParametersParser can not be null");
+    this.fhirContext = fhirContext;
+    this.fhirPath = FhirPathCache.cachedForContext(fhirContext);
+    this.fhirDal = fhirDal;
+    this.libraryProcessor = libraryProcessor;
+    this.expressionEvaluator = expressionEvaluator;
+    this.activityDefinitionProcessor = activityDefinitionProcessor;
+    this.operationParametersParser = operationParametersParser;
+  }
+
+  public IBaseParameters apply(IdType theId, String patientId, String encounterId, String practitionerId,
+      String organizationId, String userType, String userLanguage, String userTaskContext, String setting,
+      String settingContext, Boolean mergeNestedCarePlans, IBaseParameters parameters, Boolean useServerData,
+      IBaseBundle bundle, IBaseParameters prefetchData, IBaseResource dataEndpoint, IBaseResource contentEndpoint,
+      IBaseResource terminologyEndpoint) {
+
+    // warn if prefetchData exists
+    // if no data anywhere blow up
+    IBaseResource basePlanDefinition = this.fhirDal.read(theId);
+    PlanDefinition planDefinition;
+
+    if (basePlanDefinition == null) {
+      throw new IllegalArgumentException("Couldn't find PlanDefinition " + theId);
+    }
+    if (!(basePlanDefinition instanceof PlanDefinition)) {
+      throw new IllegalArgumentException(
+          "The planDefinition passed to FhirDal was " + "not a valid instance of PlanDefinition.class");
+    }
+
+    planDefinition = (PlanDefinition) basePlanDefinition;
+
+    logger.info("Performing $apply operation on PlanDefinition/" + theId);
+
+    CarePlan builder = new CarePlan();
+
+    builder.addInstantiatesCanonical(planDefinition.getIdElement().getIdPart()).setSubject(new Reference(patientId))
+        .setStatus(CarePlan.CarePlanStatus.DRAFT);
+
+    if (encounterId != null)
+      builder.setEncounter(new Reference(encounterId));
+    if (practitionerId != null)
+      builder.setAuthor(new Reference(practitionerId));
+    if (organizationId != null)
+      builder.setAuthor(new Reference(organizationId));
+    if (userLanguage != null)
+      builder.setLanguage(userLanguage);
+
+    // Each Group of actions shares a RequestGroup
+    RequestGroup requestGroup = new RequestGroup().setStatus(RequestStatus.DRAFT).setIntent(RequestIntent.PROPOSAL);
+
+    IBaseDatatype basePrefetchDataKey = operationParametersParser.getValueChild(prefetchData, "key");
```

Comment:
The "prefetchData" elements can occur multiple times, so this isn't right.

Also, they should probably just be passed straight through to the Library evaluation.